guides/installing-python.md

```diff
@@ -0,0 +1,147 @@
+
+### Remove Homebrew-installed virtualenvwrapper
+
+Before installing and configuring pyenv, uninstall virtualenvwrapper if it  is installed via Homebrew:
+
+```shell
+brew uninstall virtualenvwrapper
+```
+
+Then remove the sourcing of `virtualenvwrapper.sh` and related variables from your shell's dotfiles (`.profile`, `.bashrc` for bash users, `.zshrc` for zsh users):
+
+```shell
+export VIRTUALENVWRAPPER_PYTHON=$HOME/homebrew/bin/python
+export VIRTUALENVWRAPPER_VIRTUALENV=$HOME/homebrew/bin/virtualenv
+source $HOME/homebrew/bin/virtualenvwrapper.sh
+```
+
+### Optional: remove Homebrew-installed Python and variables
+
+Before installing and configuring pyenv, it is a good idea to uninstall all old versions of Python and virtualenvwrapper that might be installed via Homebrew, and remove any varaibles from your shell's dotfiles (`.profile`, `.bashrc` for bash users, `.zshrc` for zsh users):
+
+```shell
+brew uninstall python
+brew cleanup python
+brew uninstall virtualenvwrapper
+```
+You should also remove any variables that reference Homebrew's Python from your shell's dotfiles (`.profile`, `.bashrc` for bash users, `.zshrc` for zsh users), for example:
+
+```shell
+export PYTHONPATH=$HOME/homebrew/bin/python
+```
+
+If you do not wish to uninstall Homebrew's Python, or if you have Homebrew-installed packages that depend on Python-installed Homebrew, simply ensure that Homebrew's Python either does not exist in your shell's `PATH` variable, or that it is added to `PATH` *before* the pyenv configuration below.
+
+## Installing pyenv
+
+To [install pyenv on MacOS](https://github.com/pyenv/pyenv#homebrew-on-macos), use Homebrew:
+
+```shell
+brew update
+brew install pyenv
+brew install pyenv-virtualenv
+brew install pyenv-virtualenvwrapper
+```
+
+**Note:** pyenv-virtualenvwrapper is not strictly required, depending on how you wish to manage your virtualenvs. See the [notes in the pyenv-virtualewrapper README](https://github.com/pyenv/pyenv-virtualenvwrapper#pyenv-virtualenvwrapper). Some of our documentation assumes `virtualenvwrapper` exists, so this guide installs it. You will see a deprecation warning when using `mkvirtualenv`:
+
+```
+WARNING: the pyenv script is deprecated in favour of `python3.6 -m venv`
+```
+
+Until we switch to using Python 3 and the `venv` module full-time, this can be safely ignored.
+
+## Configuring pyenv
+
+Per the [pyenv documentation](https://github.com/pyenv/pyenv#basic-github-checkout), the follow environment variables need to be added to your shell dotfiles (`.bashrc` for bash users, `.zshrc` for zsh users).
+
+```shell
+# pyenv
+export PYENV_ROOT="$HOME/.pyenv"
+
+# pyenv-virtualenvwrapper
+export PYENV_VIRTUALENVWRAPPER_PREFER_PYVENV="true"
+export WORKON_HOME=$HOME/.virtualenvs
+
+# Ensure this placed toward the end of the file
+eval "$(pyenv init -)"
+
+# Load virtualenvwrapper
+pyenv virtualenvwrapper_lazy
+```
+
+Restart your shell.
```

Comment:
Restart your shell or `source` the file to reload it.

Reply:
`source`ing won't unset variables that might have been previously set — If someone is familiar enough in their shell, then they could probably infer what needs to happen. I'd prefer err'ing on the side of most-brute-force here.

Reply:
That's fair - full disclosure, I just restart my shell too.